docs/tools/testing/icetea.md

```diff
@@ -0,0 +1,192 @@
+
+### Installation
+
+We recommend creating a separate Python virtual environment for this purpose. To be able to run the test cases written with the Icetea test environment, you need to install Icetea in the system.
+
+#### Required dependencies:
+
+- To use **local Mbed devices**
+  - [mbed-ls](https://github.com/armmbed/mbed-ls).
+    - To enumerate connected boards.
+  - [mbed-flasher](https://github.com/ARMmbed/mbed-flasher).
+    - To flash devices automatically.
+  - Icetea installation installs both of these automatically.
+  
+**OS specific:**
+
+Linux:
+
+- `python-dev` and `python-lxml`
+`sudo apt-get install python-dev python-lxml`
+- For documentation, install sphinx:
+`apt-get install python-sphinx`
+
+OS X:
+
+- [XCode developer tools](http://osxdaily.com/2014/02/12/install-command-line-tools-mac-os-x/).
+- [lxml](http://lxml.de/installation.html#installation):
+`STATIC_DEPS=true sudo pip install lxml`
+- For documentation, install sphinx:
+`sudo port install py27-sphinx`, `sudo port select --set python python27`, `sudo port select --set sphinx py27-sphinx`
+
+Windows:
+
+- `python-lxml` installation usually requires build tools. You can, however, install it from prebuilt binaries.
+    - Download the binary for your system from the internet.
+    - Navigate to the directory where you downloaded the binary, and install it with `pip install <insert_file_name>`.
+- For documentation, install sphinx with pip:
+`pip install sphinx`
+
+#### Optional dependencies
+
+- To decorate your console log with colors, install the coloredlogs module using pip:
+`pip install coloredlogs`
+    - Windows does not support the coloredlogs module.
+- [valgrind](http://valgrind.org).
+- [gdb](https://www.gnu.org/software/gdb/).
+
+#### Installation instructions
+
+```
+git clone https://github.com/ARMmbed/icetea.git
+cd icetea
+python setup.py install
+```
+
+**Linux**
+
+To run test cases with hardware in Linux without sudo rights:
+
+```
+sudo usermod -a -G dialout username
+Log out & log in back to Linux
+```
+
+This command adds the user `username` to the `dialout` group and grants the required permissions to the USB ports.
+
+### Use
+
+To print the help page:
+
+`icetea --help`
+
+All CLI parameters are described in [icetea.md](https://github.com/ARMmbed/icetea/blob/master/doc/Icetea.md)
+
+To list all local testcases from the `./testcases` subfolder:
+
+`icetea --list`
+
+#### Test case API
+
+To execute a single command line:
+
+` execute_command(<dut>, <cmd>, (<arguments>) ) `
+
+or alias `command(...)`
+
+
+```
+dut[number]             #dut index (1..)
```

Comment:
Could you please add more context to this code snippet? I'm not sure what it does.

Reply:
They're args for the execute_command/command function.